### ISSUE TITLE
Segfault on any contact going offline (current nightly, Linux)

### ISSUE THREAD

**Issue by raisingw** [uTox/uTox#616]
Terminal output wasn't useful but core dump attached, if of any use.
[_usr_bin_utox.1000.crash.zip](https://github.com/uTox/uTox/files/704760/_usr_bin_utox.1000.crash.zip)

**Comment by GrayHatter** [comment]
this should be fixed with #622

**Comment by cebe** [comment]
here is my stack trace:

```
$ ./utox 
Can't malloc that much, you'll probably have to move or delete your history for this peer.
		Friend number AD36ABBD802748A2F66592AFDFF5504B7F261A76E4C4A3E0B0A63398A4E4F454, count 2, actual_count 6, start at 0, error size 7305513021424533504.
Can't malloc that much, you'll probably have to move or delete your history for this peer.
		Friend number 2660B7B269E04E5B4E63A45F5729A448157C889ECB553525D74386408BD79759, count 4, actual_count 4, start at 0, error size 5116089176692883457.
Can't malloc that much, you'll probably have to move or delete your history for this peer.
		Friend number 4187C24C1E596CEA862E0DFF7D7861B33E644D6F547818B7131292F2479FBE18, count 2, actual_count 9, start at 0, error size 6085770471461093376.
AL lib: (WW) alcCloseDevice: Releasing context 0x613000045400
FileTransfer:	Can't send an avatar without data
FileTransfer:	Can't send an avatar without data
Toxcore:	Error sending message... 3
FileTransfer:	Can't send an avatar without data
FileTransfer:	Unable to use current data_hash...uTox can't resume file 
FileTransfer:	Unable to save file info... uTox can't resume file 
Gtk-Message: GtkDialog mapped without a transient parent. This is discouraged.
ASAN:SIGSEGV
=================================================================
==7350==ERROR: AddressSanitizer: SEGV on unknown address 0x000000000000 (pc 0x7f0af6a5779d sp 0x7f0ae600ac30 bp 0xbebebebebebebebe T2)
    #0 0x7f0af6a5779c (/usr/lib/x86_64-linux-gnu/libasan.so.1+0x1e79c)
    #1 0x7f0af6a8d4d5 in __interceptor_free (/usr/lib/x86_64-linux-gnu/libasan.so.1+0x544d5)
    #2 0x4181e8 in ft_decon /home/cebe/dev/tox/uTox/src/file_transfers.c:85
    #3 0x41983c in kill_file /home/cebe/dev/tox/uTox/src/file_transfers.c:253
    #4 0x4198a8 in break_file /home/cebe/dev/tox/uTox/src/file_transfers.c:263
    #5 0x41b1f6 in ft_friend_offline /home/cebe/dev/tox/uTox/src/file_transfers.c:476
    #6 0x47a35d in callback_connection_status /home/cebe/dev/tox/uTox/src/tox_callbacks.c:87
    #7 0x7f0af2e26fdb in check_friend_tcp_udp /home/cebe/dev/tox/c-toxcore/toxcore/Messenger.c:912
    #8 0x7f0af2e27147 in check_friend_connectionstatus /home/cebe/dev/tox/c-toxcore/toxcore/Messenger.c:942
    #9 0x7f0af2e271ab in set_friend_status /home/cebe/dev/tox/c-toxcore/toxcore/Messenger.c:953
    #10 0x7f0af2e29895 in handle_status /home/cebe/dev/tox/c-toxcore/toxcore/Messenger.c:2090
    #11 0x7f0af2c1f51b in handle_status /home/cebe/dev/tox/c-toxcore/toxcore/friend_connection.c:360
    #12 0x7f0af2a0e431 in connection_kill /home/cebe/dev/tox/c-toxcore/toxcore/net_crypto.c:1311
    #13 0x7f0af2a0e63b in handle_data_packet_core /home/cebe/dev/tox/c-toxcore/toxcore/net_crypto.c:1376
    #14 0x7f0af2a0eca5 in handle_packet_connection /home/cebe/dev/tox/c-toxcore/toxcore/net_crypto.c:1544
    #15 0x7f0af2a10217 in udp_handle_packet /home/cebe/dev/tox/c-toxcore/toxcore/net_crypto.c:2254
    #16 0x7f0af25f0637 in networking_poll /home/cebe/dev/tox/c-toxcore/toxcore/network.c:457
    #17 0x7f0af2e2aaad in do_messenger /home/cebe/dev/tox/c-toxcore/toxcore/Messenger.c:2535
    #18 0x7f0af621e96f in tox_iterate /home/cebe/dev/tox/c-toxcore/toxcore/tox.c:371
    #19 0x47829d in toxcore_thread /home/cebe/dev/tox/uTox/src/tox.c:511
    #20 0x7f0af59b30a3 in start_thread (/lib/x86_64-linux-gnu/libpthread.so.0+0x80a3)
    #21 0x7f0af384162c in clone (/lib/x86_64-linux-gnu/libc.so.6+0xe862c)

AddressSanitizer can not provide additional info.
SUMMARY: AddressSanitizer: SEGV ??:0 ??
Thread T2 created by T0 here:
    #0 0x7f0af6a5cbba in pthread_create (/usr/lib/x86_64-linux-gnu/libasan.so.1+0x23bba)
    #1 0x5e4773 in thread /home/cebe/dev/tox/uTox/src/xlib/main.c:151
    #2 0x5e8c9e in main /home/cebe/dev/tox/uTox/src/xlib/main.c:802
    #3 0x7f0af377ab44 in __libc_start_main (/lib/x86_64-linux-gnu/libc.so.6+0x21b44)

==7350==ABORTING
```

**Comment by GrayHatter** [comment]
fixed in #627
